### PR TITLE
feat: add new CopyableAddress component, use it in user and token popup, some style alignments

### LINKED ISSUE #1294: User Pop Over - Wallet address copy
## Description

Within the user pop over, the functionality to copy a user's wallet address has not been added and currently the truncated wallet address shows but with no way to interact with it. 

Also within the token pop-over lays the same feature that requires a copy of the token address. 

image

**The following interaction states should display:**

1. Hover state in text-blue-400 on both the wallet address and copy icon.
2. A copy state in the text-success-400 on both the wallet address and copy icon.
3. The functionality should be present across all user pop over types (top, active, new etc).
4. Applies to desktop only for the over state. The copied state should be present on responsive sizes. 
5. The wallet address should copy to the users clipboard. 

## Figma

Link to Figma user pop-over component with interaction: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=14040-848332&mode=design&t=Yv1QB1wd5npVbj8m-4

Token pop-over with interaction: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=14987-344707&mode=design&t=K2iJwZlTf2q7OyzG-4

### PR DESCRIPTION
**Pull request by bassgeta**
## Description

Since we are migrating to the new copyable address component, I created an interim `CopyableAddress` component in `v5/shared` which handles copying to clipboard according to the new designs.
I did not refactor the entire codebase as to not induce any regression bugs, I just used it in the places specified.
Also reworked the token popover to look like it should in the new designs (`TokenInfoPopover` in storybook, can also be seend on the funding page, click on the tokens right-side)

**New stuff** ✨

* `CopyableAddressV2` component which handles copy/paste and styling

**Changes** 🏗

* `className` property to the `MaskedAddress` component
* reworked `TokenInfo` to look like it does in new designs
* `MaskedAddress` now uses tailwind

**Removed**

* `TokenLink` component since it was used just for the old popover

Resolves #1294

### CODE REVIEW

File: src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx
Comment (jakubcolony) [inline src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx:11]: Import order rules not followed here

File: src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx
Comment (jakubcolony) [inline src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx:15]: I don't like this prop being called children yet not accepting `React.ReactNode` or similar

File: src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx
Comment (jakubcolony) [inline src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx:10]: We need to fix that, we have to be able to override existing CSS with tailwind classes

File: src/components/shared/MaskedAddress/MaskedAddress.tsx
Comment (jakubcolony) [inline src/components/shared/MaskedAddress/MaskedAddress.tsx:4]: Empty line unnecessary

File: src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx
Comment (jakubcolony) [inline src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx:38]: Why do we need to call `splitAddress` here if we're only checking for its error? The same seems to happen in MaskedAddress

File: src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx
Comment (jakubcolony) [inline src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx:19]: I don't see this component with this variant being used anywhere. And if it is needed indeed, the button to copy address should be an independent component that this one depends on, not the other way around

File: src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx
Comment (bassgeta) [reply src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx:38]: fair, I just copied over the old component and built upon it, I'll check if I can clean this up for a bit too

File: src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx
Comment (jakubcolony) [reply src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx:10]: You could re-write the entire MaskedAddress component using tailwind I suppose? Doesn't matter if you break "old world" components

**Review by rdig — COMMENTED** [review]
I'll hold off reviewing this until you address the issues Jakub raised

File: src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx
Comment (bassgeta) [reply src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx:15]: I actually don't mind this, because it disabled you from passing in JSX components, it limits you only to string children, which isn't bad practice. Maybe it's not applicable to this component, I can just turn this into a prop instead

File: .storybook/main.css
Comment (rdig) [inline .storybook/main.css:13]: Why were these added? Does the design call for these ?

File: src/styles/main.global.css
Comment (rdig) [inline src/styles/main.global.css:16]: Same question as before, as I don't think this font is required for the new designs

**Review by rdig — CHANGES_REQUESTED** [review]
Had a quick review, code-wise seems to ok, but styling is still off.

I'll let @melyndav give the proper styling changes review

**Review by melyndav — CHANGES_REQUESTED** [review]
Hi @bassgeta, nice start on this. A few issues to resolve:

## Token pop-over

1. Border-radius is missing
2. Border colour is missing
3. Address text should match the font in Figma
4. Padding between address and copy icon doesn't match Figma
5. Icon and text should be centre aligned 
6. For native tokens, the verified icon should show. 
7. When there is a long token name, the verified icon should also show with the truncaed content)
8. Remove '*This is the colony's native token' copy 
9. Copy icon should be 16x16. Is this also the current UI's style?
10. Check icon on copy doesn't match Figma. Is it the current UI's style? 
11. Drop shadow is wrong. 
12. Hover state is missing on the view on link. Please use the following for the hover state:

```
color: var(--light-gray-900, #101828);

/* Text/Regular Medium */
font-family: Inter;
font-size: 14px;
font-style: normal;
font-weight: 500;
line-height: 20px; /* 142.857% */
```


13. Remove the 'add to wallet button'

Basically all styles of the current UI need to be removed. We are not using any part of the UI including colours, bg, dropshadows, hover states, fonts etc. 

Some new styling changes are needed:

image

1. Native token introduced (this should already be an existing pill component created by PagePro and used across the dapp)
2. HR removed
3. Token box made wider to match the user popovers 



## User popover & hamburger menu 
1. Address text should match the font in Figma
2. Padding between address and copy icon doesn't match Figma
3. Icon and text should be centre aligned 
4. Copy icon should be 16x16.
5. Check icon on copy doesn't match Figma.

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @bassgeta. Thanks for the updates. More are needed.

1. Please make the address font-weight match Figma.
2. Please make the icon size match Figma. 
3. Please match the font weight of the token popover to Figma. 

image

**Review by jakubcolony — APPROVED** [review]
Code-wise it looks good, nice work @bassgeta!

**Review by jakubcolony — APPROVED** [review]

**Review by melyndav — CHANGES_REQUESTED** [review]
Hi @bassgeta looks like the token title is being repeated 4 times. Can we fix? 

image

Also there is a slight issue on the left corner of the box radius.

image

File: src/components/shared/TokenInfoPopover/TokenInfo.tsx
Comment (rdig) [inline src/components/shared/TokenInfoPopover/TokenInfo.tsx:44]: Whaaaat ?

Comment (bassgeta) [reply src/components/shared/TokenInfoPopover/TokenInfo.tsx:44]: Sorry, Friday afternoon testing gone wrong, totally my bad :facepalm:

File: src/components/shared/TokenInfoPopover/TokenInfo.tsx
Comment (rdig) [inline src/components/shared/TokenInfoPopover/TokenInfo.tsx:48]: Screenshot from 2023-11-20 17-12-18

Hovering this should show you the token's full name (via the in built browser tooltip)

Ie: set the `title` attribute

**Review by rdig — DISMISSED** [review]
I'm mostly good with this however wait for @melyndav 's design ok before merging.

Also, rebase this branch, as it's waaaay out of date with `master`. It still has the transaction loading loop error that was fixed last week

**Review by jakubcolony — APPROVED** [review]

**Review by melyndav — APPROVED** [review]
Heya @bassgeta thanks for all the updates with this PR. approved from my end. :)

**Pull request merged**
